server/game/TerminalCondition.js

```diff
@@ -26,7 +25,7 @@ class TerminalCondition {
     }
 
     checkCondition() {
-        return this.condition() && (!this.event || this.event.cancelled);
+        return this.condition() && !this.event;
```

Comment:
why are you removing this?  Can you test Weasand with an Armageddon Cloak?

Reply:
Rolled back TerminalCondition to what it was and added test case to test Armageddon Cloak. It does not work. The creature does not leave play.